core/trino-spi/src/main/java/io/trino/spi/exchange/ExchangeManager.java

```diff
@@ -63,9 +63,10 @@
      * A new sink instance is created by the coordinator for every task attempt (see {@link Exchange#instantiateSink(ExchangeSinkHandle, int)})
      *
      * @param handle returned by {@link Exchange#instantiateSink(ExchangeSinkHandle, int)}
+     * @param requestHandleUpdate callback to request an update of {@link ExchangeSinkInstanceHandle}
```

Comment:
are there some constraints on when the callback can be called. E.g can this be called from synchronized section. Should Exchange implementation expect timely return from the callback.

Generally I think this is first occurence when in SPI we expect plugin to callback to engine. Maybe it would be better to retructure that so engine polls for information. WDYT?

Reply:
Discussed offline. For now we are not going to add a callback. Instead we will let the engine to pool the `ExchangeSink` status and update it when necessary.